server/routes/users.js

```diff
@@ -82,4 +84,43 @@ router.get('/:name/productslist/:id', async function (req, res) {
 	}
 });
 
+/*router.get('/:id', verifyToken, async (req, res) => {
+	try {
+		const user = await User.findById(req.params.id);
+		res.status(200).send(user);
+	} catch (err) {
+		console.error('User not found');
+		res.status(500);
+	}
+});*/
+
+router.put('/:id/edit',
+[
+	check('name', 'Name is required').not().isEmpty(),
+	check('password', 'Please enter a password with 6 or more characters').isLength({ min: 6 })
+],
+verifyToken, 
+async (req, res) => {
+	const errors = validationResult(req);
+	if (!errors.isEmpty()) {
+		return res.status(400).json({ errors: errors.array() });
+	}
+	const { name, password, email } = req.body;
+	try {
+		let checkUser = await User.findOne({ name });
+		if (checkUser) {
+			return res.status(400).json({ errors: [{ msg: 'Username already exists.' }] });
+		}
+		let user = await User.findOne({ email });
+		user.name = name;
+		const salt = await bcrypt.genSalt(10);
+		user.password = await bcrypt.hash(password, salt);
+		await user.save();
+		res.status(200).send(user);
+	} catch (err) {
+		console.error(err.message);
+		res.status(500).send('Server error');
+	}
+});
+
```

Comment:
Since all of this information is usually handled in auth.js, maybe this logic should be there too? To keep similiar tasks in the same place. What do you think?

Reply:
Yeah that is correct! No worries. Will add it in auth.js.